libweasyl/libweasyl/models/tables.py

```diff
@@ -391,11 +391,22 @@ def default_fkey(*args, **kwargs):
         },
     }, length=20), nullable=False, server_default=''),
     Column('email', String(length=100), nullable=False, server_default=''),
+    Column('twofa_secret', String(length=420), nullable=True),
 )
 
 Index('ind_login_login_name', login.c.login_name)
 
 
+twofa_recovery_codes = Table(
+    'twofa_recovery_codes', metadata,
+    Column('userid', Integer(), primary_key=True, nullable=False),
+    Column('recovery_code', String(length=100), primary_key=True, nullable=False),
```

Comment:
Since this is now hashed, it shouldn’t be a primary key. You may also want to name it to match (`recovery_code_hash`).